programs/cleos/main.cpp

```diff
@@ -3321,16 +3327,26 @@ int main( int argc, char** argv ) {
          chain_id = chain_id_type(str_chain_id);
       }
 
-      if( str_private_key.size() == 0 ) {
+      if( str_private_key.size() > 0 ) {
+         private_key_type priv_key;
+         try {
+            priv_key = private_key_type(str_private_key);
+         } EOS_RETHROW_EXCEPTIONS(private_key_type_exception, "Invalid private key: ${private_key}", ("private_key", str_private_key))
+         trx.sign(priv_key, *chain_id);
+      } else if( str_keosd_pub_key.size() > 0 ) {
+         public_key_type pub_key;
+         try {
+            pub_key = public_key_type(str_keosd_pub_key);
+         } EOS_RETHROW_EXCEPTIONS(public_key_type_exception, "Invalid public key: ${public_key}", ("public_key", str_keosd_pub_key))
+         fc::variant keys_var(flat_set<public_key_type>{ pub_key });
+         sign_transaction(trx, keys_var, *chain_id);
+      } else {
          std::cerr << localized("private key: ");
          fc::set_console_echo(false);
          std::getline( std::cin, str_private_key, '\n' );
          fc::set_console_echo(true);
       }
```

Comment:
Looks like you can no longer sign with a private key provided to `getline`

Reply:
Right, my bad. I forgot to use the str_private_key after refactoring the code. Thanks a lot for catching this!